Finishing up ICoredebug module4 and process11 (#31068)
dotnet/docs: docs/framework/unmanaged-api/debugging/icordebugprocess11-interface.md

---
description: "Learn more about: ICorDebugProcess11 Interface"
title: "ICorDebugProcess11 Interface"
ms.date: "06/06/2022"
api_name: 
  - "ICorDebugProcess11"
api_location: 
  - "mscordbi.dll"
api_type: 
  - "COM"
ms.assetid: 71aee5f3-5e10-44fa-be69-6d8a475f2c14
topic_type: 
  - "apiref"
---
# ICorDebugProcess11 Interface

[Supported in .NET Framework 4.5.2 and later versions.]  
  
 Provides a method that configures the debugger to handle in-memory metadata updates in the target process.  
  
## Methods  
  
|Method|Description|  
|------------|-----------------|  

> |[EnumerateLoaderHeapMemoryRegions Method](icordebugprocess11-enumerateloaderheapmemoryregions-method.md)|Sets a value that determines how the debugger handles in-memory updates to metadata within the target process.|  

Reviewer: This text still refers to the old incorrect description

---
# ICorDebugProcess11 Interface

> [Supported in .NET Core 5.0 and later versions.]  

Reviewer 2: ```suggestion
[Supported in .NET 5 and later versions.]  
```

> Provides a method that enumerates ranges of native memory that are used by the .NET runtime to store internal data structures that describe .NET types and methods. It is the same information that would be shown running the SOS "eeheap -loader" command.

Reviewer 2: ```suggestion
Provides a method that enumerates ranges of native memory that are used by the .NET runtime to store internal data structures that describe .NET types and methods. The returned information is the same that would be shown by using the SOS `eeheap -loader` command.
```

## Methods  
  
|Method|Description|  
|------------|-----------------|  

> |[EnumerateLoaderHeapMemoryRegions Method](icordebugprocess11-enumerateloaderheapmemoryregions-method.md)|enumerates ranges of native memory that are used by the .NET runtime to store internal data structures that describe .NET types and methods. It is the same information that would be shown running the SOS "eeheap -loader" command.|  

Reviewer 2: ```suggestion
|[EnumerateLoaderHeapMemoryRegions Method](icordebugprocess11-enumerateloaderheapmemoryregions-method.md)| Enumerates ranges of native memory that are used by the .NET runtime to store internal data structures that describe .NET types and methods. The information returned is the same that would be shown by using the SOS `eeheap -loader` command. |
```

 **Header:** CorDebug.idl, CorDebug.h  
  
 **Library:** CorGuids.lib  

>  **.NET Core Supported Versions:** 5.0 and above.  

Reviewer 3: ```suggestion
 **.NET Versions:** [!INCLUDE[net_current_v452plus](../../../../includes/net-core-50-md.md)]
```

>  **.NET Versions:** [!INCLUDE[net_current_v452plus](../../../../includes/net-core-50-md.md)]  

Reviewer 3: ```suggestion
 **.NET Versions:** [!INCLUDE[net-core-50-plus](../../../../includes/net-core-50-md.md)]  
```
